pkg/controller/backup/backup_controller.go

```diff
@@ -186,31 +186,63 @@ func (c *Controller) updateBackup(cur interface{}) {
 		if newBackup.Spec.Mode == v1alpha1.BackupModeLog {
 			return
 		}
-		selector, err := label.NewBackup().Instance(newBackup.GetInstanceName()).BackupJob().Backup(name).Selector()
-		if err != nil {
-			klog.Errorf("Fail to generate selector for backup %s/%s, %v", ns, name, err)
-			return
-		}
-		pods, err := c.deps.PodLister.Pods(ns).List(selector)
-		if err != nil {
-			klog.Errorf("Fail to list pod for backup %s/%s with selector %s, %v", ns, name, selector, err)
+
+		// backup retry after it has been scheduled, and scheduled means already create k8s job
```

Comment:
Can you simplify this function by using top-down method, make it easy to be understood? I don't understand why we execute these logic under this condition branch. Can you make your code self-explanation？

Reply:
please split this function into smalls

Reply:
What logic is for Snapshot retry what logic is for other BR mode？ Can you make it easy to see the logic path?

Reply:
current logic is to make it easy to realize other kinds backup restart. because they are similar, but we just realize snapshot now.

Reply:
the logic is typical reentrant logic, check and then run:
1. if retry after schedule --> clean job before add queue
2. if there is no record about failed, check is there a failed, if true, record it
3. after detect failed, retry according to backoff limits, if it meet limit, retry failed.

Reply:
i left main logic in this `updateBackup` which is the  entry of retry, it's to make it easy to understand, but maybe the function name and nodes is not good enough, i will optimize them.

Reply:
after refine code, i add notes explain the main logic.
https://github.com/pingcap/tidb-operator/pull/4895/files#diff-0027249f0ff2488fc43bf3a1503d2353f0466813f6db16d6ee6179256fe144e4R431